lib/Connection.php

```diff
@@ -249,7 +256,7 @@ private function closeSocket() {
         $this->outputBuffer = "";
         $this->outputBufferLength = 0;
 
-        if (is_resource($this->socket)) {
+        if (!$this->persistent && is_resource($this->socket)) {
```

Comment:
Correct. That's why fclose() should not be executed in destructor, only when you are explicitly closing it via a close() call.